### PR TITLE
Cleanups to Annotation.

### PR DESCRIPTION
**Pull request by anntzer**
- Reorder logic in _get_xy_transform to make the case of xy0 not being specified (i.e. invalid bbox_name) fail more obviously.
- Document "offset fontsize" option.

See discussion at #25907.

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-docstrings) guidelines

### CODE REVIEW

File: lib/matplotlib/text.py [hunk 1]
```diff
@@ -1701,15 +1693,15 @@ def transform(renderer) -> Transform
...
 
-            =================   =========================================
+            =================   =================================================
             Value               Description
-            =================   =========================================
+            =================   =================================================
             'offset points'     Offset (in points) from the *xy* value
             'offset pixels'     Offset (in pixels) from the *xy* value
-            =================   =========================================
+            'offset fontsize'   Offset (relative to fontsize) from the *xy* value
```
[the remark this answers is not in the document]
Comment (anntzer) [reply lib/matplotlib/text.py]: What other font size there is?  This is literally in the docstring for *annotate()* which is about putting a *text* somewhere.
I did remove the parentheses, though.
I do think keeping the *same* header as for xycoords is better, as this table is supposed to be a continuation of the one just above.

[the remark this answers is not in the document]
Comment (anntzer) [reply lib/matplotlib/text.py]: If your font size is 12pt, then 1 means "12 pt".

**Comment by oscargus** [comment]
`fontsize` can be used for `figure`, `subfigure` and `axes` as well? Should it be a follow-up PR to add docs for those or will you handle it here as well?

Also, I think it would be nice to a add a test for some `fontsize` variant, although I understand that nothing has changed.

Is `scale(1)` an expensive op? If not, one may consider setting the scale factor in the if-clauses.

**Comment by anntzer** [comment]
> fontsize can be used for figure, subfigure and axes as well?

I guess so, but I don't think e.g. `axes fontsize` (some number of fontsizes away from the lower left corner of the axes) is really worth documenting: for the lower left corner it is equivalent to `annotate(xy=(0, 0), xycoords="axes fraction", xytext=some_offset, textcoords="offset font")`, but the latter form also works for the other corners whereas there's no way to specify the other corners with `axes fontsize`.  So I'd say documenting it is just an attractive nuisance.  (We could rework the code to *disable* that use, but I'm not going to bother doing so here.)

> Is scale(1) an expensive op? If not, one may consider setting the scale factor in the if-clauses.

If you're thinking `tr = Affine2D().scale(... if unit == ... else ... if unit == ...)` etc., I'd say (even though I usually like that kind of constructs) that this is worse 1) because of the star-unpacking in the "fraction" case and 2) because of the need to throw ValueError in the last case (I've always wanted python to have "raise expressions" so that one can write `foo = bar if baz else raise Exception(...)`, but heh...)

**Comment by oscargus** [comment]
More something like:
```
        if unit == "points":
            scale = self.figure.dpi / 72  # dpi/72 dots per point
        elif unit == "pixels":
            scale = 1.
        ...

        return Affine2D().scale(scale).translate(*xy0)
```

**Comment by anntzer** [comment]
But there's the unit == "fraction" case, which is a bit ugly to also handle here.

**Comment by oscargus** [comment]
> But there's the unit == "fraction" case, which is a bit ugly to also handle here.

Fair enough. Didn't see that.

**Pull request merged**